### PR TITLE
Ruby solution checking all possible winning rows

### PR DESCRIPTION
**Pull request by brehberg**
# Language

What language was used for the solution?
Ruby

# Challenge

What is the name of the challenge being completed?
2021-09-20-tic-tac-toe-winner

### CODE REVIEW

File: challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  [3,5,7]
+].freeze
+
+def who_wins(board)
+  board_state = []
+  
+  # populate the board state
+  board.each.with_index(1) do |move, index|
+    row, col = move
+    board_state[row*3+col+1] = index.odd? ? 'A' : 'B'
```
Comment (parhaml) [inline challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb:20]: Nice. Take advantage of arbitrary index assigning

File: challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  # check for winning rows
+  POSSIBLE_ROWS.each do |row|
+    i, j, k = row
+    if (board_state[i] == board_state[j] and
+        board_state[i] == board_state[k])          
+          if board_state[i] != nil
```
Comment (parhaml) [inline challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb:28]: idiomatic would be 

`unless board_state[i].nil?`

**Review by parhaml — APPROVED** [review]

File: challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb (hunk 2 above)
Comment (brehberg) [reply challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb:28]: Understood.  I will make that adjustment now

**Comment by brehberg** [comment]
Hmm, I think I did the wrong thing here when trying to implement your previous advice @parhaml

File: challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    [3,5,7]
+  ].freeze
+  
+  def who_wins(board)
+    state = []
+    
+    # populate the board state
+    board.each.with_index(1) do |move, index|
+      row, col = move
+      state[row*3+col+1] = index.odd? ? 'A' : 'B'
```
Comment (xanderyzwich) [inline challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb:19]: I love the manual handling of 2d indexing to 1d.

Comment (brehberg) [reply challenges/2021-09-20-tic-tac-toe-winner/solutions/ruby/brehberg/brehberg.rb:19]: It was much easier for me to enter the possible winning rows that way. Starting with index 1 makes my brain happier. 😅

**Review by parhaml — APPROVED** [review]

**Review by xanderyzwich — APPROVED** [review]

**Pull request merged**